yarn-project/end-to-end/src/e2e_offchain_message.test.ts

```diff
@@ -0,0 +1,42 @@
+import { type AccountWalletWithSecretKey, AztecAddress, Fr } from '@aztec/aztec.js';
+import { OffchainMessageContract } from '@aztec/noir-test-contracts.js/OffchainMessage';
+
+import { jest } from '@jest/globals';
+
+import { ensureAccountsPubliclyDeployed, setup } from './fixtures/utils.js';
+
+const TIMEOUT = 120_000;
+
+describe('e2e_offchain_message', () => {
+  let contract: OffchainMessageContract;
+  jest.setTimeout(TIMEOUT);
+
+  let wallets: AccountWalletWithSecretKey[];
+  let teardown: () => Promise<void>;
+
+  beforeAll(async () => {
+    ({ teardown, wallets } = await setup(2));
+    await ensureAccountsPubliclyDeployed(wallets[0], wallets.slice(0, 2));
+    contract = await OffchainMessageContract.deploy(wallets[0]).send().deployed();
+  });
+
+  afterAll(() => teardown());
+
+  it('should emit offchain message', async () => {
+    const message = [Fr.random(), Fr.random(), Fr.random(), Fr.random(), Fr.random()];
+    const recipient = await AztecAddress.random();
+
+    const numRecursions = 3;
+
+    const provenTx = await contract.methods
+      .emit_offchain_message_for_recipient(message, recipient, numRecursions)
+      .prove();
```

Comment:
Calling prove here instead of send as I have not yet figured out the API for send method.